sdk-lib/src/main/rust/lib.rs

```diff
@@ -9,60 +9,61 @@ use std::str::FromStr;
 
 use android_logger::Config;
 use failure::format_err;
+use hdwallet::traits::{Deserialize, Serialize};
 use jni::{
     objects::{JClass, JString},
     sys::{jboolean, jbyteArray, jint, jlong, jobjectArray, jstring, JNI_FALSE, JNI_TRUE},
     JNIEnv,
 };
 use log::Level;
-use secp256k1::key::{PublicKey, SecretKey};
-use zcash_client_backend::data_api::wallet::{shield_funds, ANCHOR_OFFSET};
+use secp256k1::PublicKey;
+use zcash_client_backend::keys::UnifiedSpendingKey;
 use zcash_client_backend::{
     address::RecipientAddress,
     data_api::{
         chain::{scan_cached_blocks, validate_chain},
         error::Error,
-        wallet::{create_spend_to_address, decrypt_and_store_transaction},
-        WalletRead,
+        wallet::{
+            create_spend_to_address, decrypt_and_store_transaction, shield_transparent_funds,
+        },
+        WalletRead, WalletReadTransparent, WalletWrite, WalletWriteTransparent,
     },
     encoding::{
-        decode_extended_full_viewing_key, decode_extended_spending_key,
-        encode_extended_full_viewing_key, encode_extended_spending_key, encode_payment_address,
-        AddressCodec,
+        decode_extended_spending_key, encode_extended_full_viewing_key,
+        encode_extended_spending_key, encode_payment_address, AddressCodec,
     },
-    keys::{
-        derive_public_key_from_seed, derive_secret_key_from_seed,
-        derive_transparent_address_from_public_key, derive_transparent_address_from_secret_key,
-        spending_key, Wif,
-    },
-    wallet::{AccountId, OvkPolicy, WalletTransparentOutput},
+    keys::{sapling, UnifiedFullViewingKey},
+    wallet::{OvkPolicy, WalletTransparentOutput},
 };
+#[allow(deprecated)]
+use zcash_client_sqlite::wallet::{delete_utxos_above, get_rewind_height};
 use zcash_client_sqlite::{
     error::SqliteClientError,
     wallet::init::{init_accounts_table, init_blocks_table, init_wallet_db},
-    wallet::{
-        delete_utxos_above, get_rewind_height, put_received_transparent_utxo, rewind_to_height,
-    },
     BlockDb, NoteId, WalletDb,
 };
 use zcash_primitives::consensus::Network::{MainNetwork, TestNetwork};
+#[allow(deprecated)]
+use zcash_primitives::legacy::keys::{pubkey_to_address, AccountPrivKey};
 use zcash_primitives::{
     block::BlockHash,
     consensus::{BlockHeight, BranchId, Network, Parameters},
-    legacy::TransparentAddress,
+    legacy::{Script, TransparentAddress},
     memo::{Memo, MemoBytes},
     transaction::{
-        components::{Amount, OutPoint},
+        components::{Amount, OutPoint, TxOut},
         Transaction,
     },
-    zip32::ExtendedFullViewingKey,
+    zip32::{AccountId, DiversifierIndex, ExtendedFullViewingKey},
 };
 use zcash_proofs::prover::LocalTxProver;
 
 use crate::utils::exception::unwrap_exc_or;
 
 mod utils;
 
+const ANCHOR_OFFSET: u32 = 10;
```

Comment:
This is not what we want long-term (see draft ZIP 315), but it will do for now.

Reply:
Yep, this is intentionally a no-op in this PR.